src/QMCWaveFunctions/SPOSet.h

```diff
@@ -117,10 +117,14 @@ class SPOSet : public QMCTraits
     APP_ABORT(o.str());
   }
   /// reset parameters to the values from optimizer
-  virtual void resetParameters(const opt_variables_type& optVariables) = 0;
+  virtual void resetParameters(const opt_variables_type& optVariables) {}
 
-  /// check in/out parameters to the global list of parameters used by the optimizer
+  /** check in parameters to the global list of parameters used by the optimizer.
+   * This is a query function and should never be implemented as a feature blocker.
```

Comment:
I don't quite understand what's happening by looking at this comment and just the PR.  `resetParameters` is the function that changed from needing to be implemented by all the derived classes to one with a default implementation. `checkInVariables` already has a default implementation.  So this comment references some change that isn't in this PR?

Reply:
[LCAOrbitalSet.h](https://github.com/QMCPACK/qmcpack/pull/4136/files#diff-3008c4a5fdfa852f307be230c2d1e896b22871f8a9bf40fc2cdfd29be81da657)
I removed the blocker. So I added this comment to explain why.
I don't see a good reason to require implementing empty reset parameters.
There is a bigger picture I'm moving towards. https://github.com/ye-luo/qmcpack/wiki/Optimization-API-changes
It is still quite preliminary.

Reply:
I missed the removal in LCAOrbitalSet. D'oh.